### PR TITLE
code refactoring to allow the ability to provide a custom ProxyServle…

### PR DESCRIPTION
**Pull request by hitejinder**
Changes
====
- code refactoring to allow the ability to provide a custom ProxyServletImpl

We need this at Lyft to be able to replace the schema refresh query with another query that goes to an ETL table instead of hivemetastore. This makes the query finish a lot faster.

### CODE REVIEW

**Comment by hitejinder** [comment]
/ptal (👀) @ssanthanam185 @bearcage @Htiango @endoplasmicR @nishantrayan @JamesRTaylor

**Comment by hitejinder** [comment]
meow @ssanthanam185 @bearcage @Htiango @endoplasmicR @nishantrayan @JamesRTaylor

File: proxyserver/src/main/java/com/lyft/data/proxyserver/ProxyServer.java [hunk 1]
```diff
@@ -27,12 +27,19 @@
...
 
   public ProxyServer(ProxyServerConfiguration config, ProxyHandler proxyHandler) {
+    this(config, proxyHandler, new ProxyServletImpl());
+  }
+
+  public ProxyServer(ProxyServerConfiguration config, ProxyHandler proxyHandler,
+                     ProxyServletImpl proxy) {
     this.server = new Server();
     this.server.setStopAtShutdown(true);
+    this.proxy = proxy;
```
Comment (endoplasmicR) [inline proxyserver/src/main/java/com/lyft/data/proxyserver/ProxyServer.java:42]: Do we need to reset this object? is this object reusable?

**Comment by endoplasmicR** [comment]
👍

**Comment by JamesRTaylor** [comment]
👍

**Comment by nishantrayan** [comment]
👍

**Pull request merged**